sdk/storage/Azure.Storage.Files/src/Models/FileSmbProperties.cs

```diff
@@ -0,0 +1,166 @@
+            this.FilePermissionKey = rawStorageFileProperties.FilePermissionKey;
+            this.FileCreationTime = rawStorageFileProperties.FileCreationTime;
+            this.FileLastWriteTime = rawStorageFileProperties.FileLastWriteTime;
+            this.FileChangeTime = rawStorageFileProperties.FileChangeTime;
+            this.FileId = rawStorageFileProperties.FileId;
+            this.ParentId = rawStorageFileProperties.FileParentId;
+        }
+
+        internal FileSmbProperties(FlattenedStorageFileProperties flattenedStorageFileProperties)
+        {
+            this.FileAttributes = CloudFileNtfsAttributesHelper.ToAttributes(flattenedStorageFileProperties.FileAttributes);
+            this.FilePermissionKey = flattenedStorageFileProperties.FilePermissionKey;
+            this.FileCreationTime = flattenedStorageFileProperties.FileCreationTime;
+            this.FileLastWriteTime = flattenedStorageFileProperties.FileLastWriteTime;
+            this.FileChangeTime = flattenedStorageFileProperties.FileChangeTime;
+            this.FileId = flattenedStorageFileProperties.FileId;
+            this.ParentId = flattenedStorageFileProperties.FileParentId;
+        }
+
+        internal FileSmbProperties(RawStorageDirectoryInfo RawStorageDirectoryInfo)
+        {
+            this.FileAttributes = CloudFileNtfsAttributesHelper.ToAttributes(RawStorageDirectoryInfo.FileAttributes);
+            this.FilePermissionKey = RawStorageDirectoryInfo.FilePermissionKey;
+            this.FileCreationTime = RawStorageDirectoryInfo.FileCreationTime;
+            this.FileLastWriteTime = RawStorageDirectoryInfo.FileLastWriteTime;
+            this.FileChangeTime = RawStorageDirectoryInfo.FileChangeTime;
+            this.FileId = RawStorageDirectoryInfo.FileId;
+            this.ParentId = RawStorageDirectoryInfo.FileParentId;
+        }
+
+        internal FileSmbProperties(RawStorageDirectoryProperties rawStorageDirectoryProperties)
+        {
+            this.FileAttributes = CloudFileNtfsAttributesHelper.ToAttributes(rawStorageDirectoryProperties.FileAttributes);
+            this.FilePermissionKey = rawStorageDirectoryProperties.FilePermissionKey;
+            this.FileCreationTime = rawStorageDirectoryProperties.FileCreationTime;
+            this.FileLastWriteTime = rawStorageDirectoryProperties.FileLastWriteTime;
+            this.FileChangeTime = rawStorageDirectoryProperties.FileChangeTime;
+            this.FileId = rawStorageDirectoryProperties.FileId;
+            this.ParentId = rawStorageDirectoryProperties.FileParentId;
+        }
+
+        /// <summary>
+        /// Checks if two FileSmbProperties are equal.
+        /// </summary>
+        /// <param name="other"></param>
+        /// <returns></returns>
+        public override bool Equals(object other)
+            => other is FileSmbProperties && this.Equals((FileSmbProperties)other);
```

Comment:
You can use the modern syntax to avoid the cast:

```C#
            => other is FileSmbProperties props && this.Equals(props);
```

Reply:
fixed